src/esw/brushed_motors/Controller.cpp

```diff
@@ -196,12 +198,43 @@ void Controller::turnOn() const {
 }
 
 // REQUIRES: nothing
-// MODIFIES: isLive
+// MODIFIES: nothing
+// EFFECTS: Returns a combined ID for both the deviceAddress and motorID
+// MotorID can only be max 3 bits (0-5), and device address is max 2 bits (1 or 2)
+uint8_t Controller::combineDeviceMotorID() const {
+    return (deviceAddress << 3) | motorID;
+}
+
+// REQUIRES: nothing
+// MODIFIES: liveMap
 // EFFECTS: I2C bus, if not already live,
 // configures the physical controller.
 // Then makes live.
+
+std::unordered_map<uint8_t, std::string> Controller::liveMap;
+std::mutex Controller::liveMapLock;
+
 void Controller::makeLive() {
-    if (isLive) {
+    // if (isLive) {
+    //     return;
+    // }
+
+    uint8_t key = combineDeviceMotorID();
+
+    // if key is absent, no motor with that key has been made live
+    auto it = liveMap.find(key);
+    if (it == liveMap.end()) {
+        // Map entry starts with an empty string until that joint is live
+        std::string jointName;
+        liveMap.emplace(key, jointName);
```

Comment:
Is it possible to just pass an empty string to `emplace()`? Maybe this causes some error but the whole point of emplace is to avoid copying string objects like this.

Reply:
Also, you shouldn't need to call find() again after this, since emplace returns a pair with the iterator as the first half (https://cplusplus.com/reference/unordered_map/unordered_map/emplace/)